src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarWriter.cs

```diff
@@ -126,16 +128,36 @@ public void WriteEntry(string fileName, string? entryName)
             ReadFileFromDiskAndWriteToArchiveStreamAsEntry(fullPath, entryName);
         }
 
-        // /// <summary>
-        // /// Asynchronously writes the specified file into the archive stream as a tar entry.
-        // /// </summary>
-        // /// <param name="fileName">The path to the file to write to the archive.</param>
-        // /// <param name="entryName">The name of the file as it should be represented in the archive. It should include the optional relative path and the filename.</param>
-        // /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
-        // public Task WriteEntryAsync(string fileName, string? entryName, CancellationToken cancellationToken = default)
-        // {
-        //     throw new NotImplementedException();
-        // }
+        /// <summary>
+        /// Asynchronously writes the specified file into the archive stream as a tar entry.
+        /// </summary>
+        /// <param name="fileName">The path to the file to write to the archive.</param>
+        /// <param name="entryName">The name of the file as it should be represented in the archive. It should include the optional relative path and the filename.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None" />.</param>
+        /// <returns>A task that represents the asynchronous write operation.</returns>
+        /// <exception cref="ObjectDisposedException">The archive stream is disposed.</exception>
+        /// <exception cref="ArgumentException"><paramref name="fileName"/> or <paramref name="entryName"/> is <see langword="null"/> or empty.</exception>
+        /// <exception cref="IOException">An I/O problem occurred.</exception>
+        public async Task WriteEntryAsync(string fileName, string? entryName, CancellationToken cancellationToken = default)
```

Comment:
@stephentoub this public API returns a `Task`. My immediate assumption was that all the internal async APIs this method calls should also return a `Task`. Is there anything preventing me from changing all the internal methods to return a `ValueTask` instead?

Reply:
> Is there anything preventing me from changing all the internal methods to return a ValueTask instead?

No, but just making that change will typically only yield benefits in the case of `Task<T>` to `ValueTask<T>`.